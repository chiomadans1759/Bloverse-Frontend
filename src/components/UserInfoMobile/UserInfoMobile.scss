#user-info-mobile {
  padding: 0;
}
