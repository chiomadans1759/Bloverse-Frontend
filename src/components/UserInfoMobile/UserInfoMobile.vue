<template>
  <main id="user-info-mobile" class="container d-block d-sm-none">
    <profile :hideMobileUserInfo="hideMobileUserInfo"
    displayInterest hideFollowing displayFollowers />
    <country-card v-if="!hideCountry" />
    <category-card v-if="!displayInterest" />
  </main>
</template>

<script>
import CountryCard from '@/components/CountryCard/CountryCard';
import Profile from '@/components/ProfileCard/ProfileCard';
import CategoryCard from '@/components/CategoryCard/CategoryCard';

export default {
  name: 'UserInfoMobile',
  components: { Profile, CountryCard, CategoryCard },
  props: {
    hideMobileUserInfo: {
      type: Function,
      required: false,
      default: () => {},
    },
    displayInterest: {
      type: Boolean,
      default: false,
    },
    hideFollowing: {
      type: Boolean,
      default: false,
    },
    displayFollowers: {
      type: Boolean,
      default: false,
    },
    hideCountry: {
      type: Boolean,
      default: false
    },
  }
}
</script>

<style lang="scss" scoped>
  @import './UserInfoMobile.scss';
</style>
