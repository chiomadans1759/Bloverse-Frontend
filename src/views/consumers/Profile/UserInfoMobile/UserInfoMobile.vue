<template>
  <main id="user-info-mobile" class="container d-block d-sm-none">
    <profile :hideMobileUserInfo="hideMobileUserInfo" />
    <country-card />
    <category-card />
  </main>
</template>

<script>
import CountryCard from '@/components/CountryCard/CountryCard';
import Profile from '@/components/ProfileCard/ProfileCard';
import CategoryCard from '@/components/CategoryCard/CategoryCard';

export default {
  name: 'UserInfoMobile',
  components: { Profile, CountryCard, CategoryCard },
  props: {
    hideMobileUserInfo: {
      type: Function,
      required: false,
      default: () => {},
    }
  }
}
</script>

<style lang="scss" scoped>
  @import './UserInfoMobile.scss';
</style>

