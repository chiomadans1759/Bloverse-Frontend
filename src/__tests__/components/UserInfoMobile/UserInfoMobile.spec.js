import { mount, RouterLinkStub } from '@vue/test-utils';
import UserInfoMobile from '@/components/UserInfoMobile/UserInfoMobile.vue';

describe('CountryCard Component', () => {
  it('should render correctly', () => {

    const wrapper = mount(UserInfoMobile, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
