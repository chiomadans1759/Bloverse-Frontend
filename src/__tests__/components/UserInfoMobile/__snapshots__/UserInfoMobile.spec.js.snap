// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryCard Component should render correctly 1`] = `
<main
  class="container d-block d-sm-none"
  id="user-info-mobile"
>
  <main
    class="card profile"
  >
    <img
      alt="..."
      class="card-img-top"
      src="@/assets/Avatar.png"
    />
     
    <div
      class="d-flex justify-content-between p-3"
    >
      <span
        class="d-inline d-sm-none bl-back-arrow"
      >
        <i
          class="fal fa-long-arrow-left fa-2x back-arrow"
        />
      </span>
    </div>
     
    <div
      class="d-flex justify-content-end profile-edit-icon pr-2"
    >
      <i
        class="fal fa-user-edit fa-2x"
      />
    </div>
     
    <div
      class="avatar mx-4"
    >
      <img
        alt="..."
        class="profile-img rounded-circle"
        src="@/assets/Avatar2.png"
      />
       
      <span
        class="profile-name pl-3"
      >
        Username
      </span>
    </div>
     
    <div
      class="card-body profile-body"
    >
      <h5
        class="card-title"
      >
        Full Name
      </h5>
       
      <p
        class="contact"
      >
        <span
          class="pr-2 pb-5"
        >
          <i
            class="fal fa-phone"
          />
        </span>
         +234 000 000 0000
    
      </p>
       
      <p>
        <span
          class="pr-2 py-1"
        >
          <i
            class="fal fa-envelope"
          />
        </span>
         email@bloverse.com
    
      </p>
       
      <p>
        <span
          class="pr-2"
        >
          <i
            class="fal fa-map-marker-alt"
          />
        </span>
         Lagos, Nigeria
    
      </p>
       
      <div
        class="py-4 profile-content"
      >
        <p
          class="profile__text"
        >
          Some quick example text to build on the card title and make up the bulk of the card's content.
        </p>
      </div>
       
      <!---->
       
      <div
        class="follow d-flex align-items-center"
      >
        <a
          data-toggle=""
          href="#"
        >
          Followers
        </a>
         
        <span
          class="badge followers-icon badge-primary ml-2"
        >
          5
        </span>
      </div>
    </div>
     
    <div
      class="interests"
    >
      <main
        class="category-card p-4 p-sm-3"
      >
        <div
          class="header d-flex justify-content-between "
        >
          <h5
            class=""
          >
            <strong>
              Interest
            </strong>
          </h5>
           
          <span
            class="edit-icon"
          >
            <i
              class="fal fa-edit mr-2"
            />
          </span>
        </div>
         
        <div
          class="mt-1"
        >
          <!---->
           
          <div
            class="category-list d-flex flex-wrap mt-2"
          >
            <div
              class="category border rounded my-1 mr-2 p-2"
            >
              <span>
                Category One
              </span>
            </div>
             
            <div
              class="category border rounded my-1 mr-2 p-2"
            >
              <span>
                Category Two
              </span>
            </div>
             
            <div
              class="category border rounded my-1 mr-2 p-2"
            >
              <span>
                Category Three
              </span>
            </div>
             
            <div
              class="category border rounded my-1 mr-2 p-2"
            >
              <span>
                Category Four
              </span>
            </div>
             
            <div
              class="category border rounded my-1 mr-2 p-2"
            >
              <span>
                Category Five
              </span>
            </div>
             
            <div
              class="category border rounded my-1 mr-2 p-2"
            >
              <span>
                Category Six
              </span>
            </div>
             
            <div
              class="category border rounded my-1 mr-2 p-2"
            >
              <span>
                Category Seven
              </span>
            </div>
          </div>
        </div>
      </main>
    </div>
     
    <div
      class="social p-3"
    >
      <div
        class="d-flex justify-content-between"
      >
        <h6
          class="mb-3"
        >
          <strong>
            Social Media
          </strong>
        </h6>
         
        <span
          class="edit-icon"
        >
          <i
            class="fal fa-edit mr-2"
          />
        </span>
      </div>
       
      <div
        class="d-flex flex-wrap"
      >
        <div
          class="mr-2"
        >
          <div
            class="border rounded p-2 mb-1 d-flex align-items-center"
          >
            <img
              alt="facebook"
              class="social-icon"
              src="@/assets/facebook-square.svg"
            />
             
            <h6
              class="social-text text-center pl-2"
            >
              Facebook
            </h6>
          </div>
        </div>
         
        <div
          class="mr-2"
        >
          <div
            class="border rounded p-2 mb-1 d-flex align-items-center"
          >
            <img
              alt="facebook"
              class="social-icon"
              src="@/assets/twitter-square.svg"
            />
             
            <h6
              class="social-text text-center pl-2"
            >
              Twitter
            </h6>
          </div>
        </div>
         
        <div
          class="mr-2"
        >
          <div
            class="border rounded p-2 mb-1 d-flex align-items-center"
          >
            <img
              alt="facebook"
              class="social-icon"
              src="@/assets/instagram-square.svg"
            />
             
            <h6
              class="social-text text-center pl-2"
            >
              Instagram
            </h6>
          </div>
        </div>
         
        <div
          class="mr-2"
        >
          <div
            class="border rounded p-2 mb-1 d-flex align-items-center"
          >
            <img
              alt="facebook"
              class="social-icon"
              src="@/assets/linkedin-square.svg"
            />
             
            <h6
              class="social-text text-center pl-2"
            >
              Linkedin
            </h6>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="card-footer text-muted bg-white footer-color"
    >
      
    Member Since:
    
      <span
        class="text-dark"
      >
        27 December, 2018
      </span>
    </div>
     
    <main
      id="bl-modal"
    >
      <div
        aria-hidden="true"
        aria-labelledby="customModal"
        class="modal fade"
        id="following-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog modal-dialog-centered"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header"
            >
              <h5
                class="modal-title"
                id="customModal"
              >
                Following
              </h5>
               
              <div>
                <!---->
                 
                <!---->
              </div>
               
              <button
                aria-label="Close"
                class="close"
                data-dismiss="modal"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  Ã—
                </span>
              </button>
            </div>
             
            <div
              class="modal-body"
            >
              <div
                class="bl-modal-content"
              >
                <div
                  class="row align-items-center mt-2 mb-2"
                >
                  <div
                    class="col-3 col-md-2 d-flex justify-content-center"
                  >
                    <img
                      class="user-image img-fluid"
                      src="@/assets/smiling-author.jpg"
                    />
                  </div>
                   
                  <div
                    class="col-9 col-sm-9 col-md-6 pl-0"
                  >
                    <p
                      class="username"
                    >
                      
            JohnDoe
            
                      <span
                        class="sub-span point d-inline d-md-none"
                      >
                        <a
                          href="#"
                        >
                          Unfollow
                        </a>
                      </span>
                    </p>
                     
                    <p
                      class="sub-text d-flex flex-wrap"
                    >
                      <span>
                        <span
                          class="sub-span"
                        >
                          24
                        </span>
                         Followers
            
                      </span>
                       
                      <span>
                        <span
                          class="sub-span point"
                        >
                          234
                        </span>
                         Articles
            
                      </span>
                    </p>
                  </div>
                   
                  <div
                    class="d-none d-sm-none col-md-4 d-md-flex justify-content-end"
                  >
                    <button
                      class="btn btn-outline-primary"
                      type="button"
                    >
                      Following
                    </button>
                  </div>
                </div>
                 
                <div
                  class="row align-items-center mt-2 mb-2"
                >
                  <div
                    class="col-3 col-md-2 d-flex justify-content-center"
                  >
                    <img
                      class="user-image img-fluid"
                      src="@/assets/smiling-author.jpg"
                    />
                  </div>
                   
                  <div
                    class="col-9 col-sm-9 col-md-6 pl-0"
                  >
                    <p
                      class="username"
                    >
                      
            JohnDoe
            
                      <span
                        class="sub-span point d-inline d-md-none"
                      >
                        <a
                          href="#"
                        >
                          Unfollow
                        </a>
                      </span>
                    </p>
                     
                    <p
                      class="sub-text d-flex flex-wrap"
                    >
                      <span>
                        <span
                          class="sub-span"
                        >
                          24
                        </span>
                         Followers
            
                      </span>
                       
                      <span>
                        <span
                          class="sub-span point"
                        >
                          234
                        </span>
                         Articles
            
                      </span>
                    </p>
                  </div>
                   
                  <div
                    class="d-none d-sm-none col-md-4 d-md-flex justify-content-end"
                  >
                    <button
                      class="btn btn-outline-primary"
                      type="button"
                    >
                      Following
                    </button>
                  </div>
                </div>
                 
                <div
                  class="row align-items-center mt-2 mb-2"
                >
                  <div
                    class="col-3 col-md-2 d-flex justify-content-center"
                  >
                    <img
                      class="user-image img-fluid"
                      src="@/assets/smiling-author.jpg"
                    />
                  </div>
                   
                  <div
                    class="col-9 col-sm-9 col-md-6 pl-0"
                  >
                    <p
                      class="username"
                    >
                      
            JohnDoe
            
                      <span
                        class="sub-span point d-inline d-md-none"
                      >
                        <a
                          href="#"
                        >
                          Unfollow
                        </a>
                      </span>
                    </p>
                     
                    <p
                      class="sub-text d-flex flex-wrap"
                    >
                      <span>
                        <span
                          class="sub-span"
                        >
                          24
                        </span>
                         Followers
            
                      </span>
                       
                      <span>
                        <span
                          class="sub-span point"
                        >
                          234
                        </span>
                         Articles
            
                      </span>
                    </p>
                  </div>
                   
                  <div
                    class="d-none d-sm-none col-md-4 d-md-flex justify-content-end"
                  >
                    <button
                      class="btn btn-outline-primary"
                      type="button"
                    >
                      Following
                    </button>
                  </div>
                </div>
              </div>
            </div>
             
            <!---->
          </div>
        </div>
      </div>
    </main>
  </main>
   
  <!---->
   
  <!---->
</main>
`;
